test/dotnet.Tests/CommandTests/Run/RunFileTests.cs

```diff
@@ -1447,6 +1459,84 @@ Hello from Program
                 """);
     }
 
+    /// <summary>
+    /// <c>Properties/launchSettings.json</c> takes precedence over <c>Program.run.json</c>.
+    /// </summary>
+    [Fact]
+    public void LaunchProfile_Precedence()
+    {
+        var testInstance = _testAssetsManager.CreateTestDirectory();
+        File.WriteAllText(Path.Join(testInstance.Path, "Program.cs"), s_program + """
+
+            Console.WriteLine($"Message: '{Environment.GetEnvironmentVariable("Message")}'");
+            """);
+        Directory.CreateDirectory(Path.Join(testInstance.Path, "Properties"));
+        File.WriteAllText(Path.Join(testInstance.Path, "Properties", "launchSettings.json"), s_launchSettings.Replace("TestProfileMessage", "PropertiesLaunchSettingsJson"));
+        File.WriteAllText(Path.Join(testInstance.Path, "Program.run.json"), s_launchSettings.Replace("TestProfileMessage", "ProgramRunJson"));
```

Comment:
I am curious if this is a situation that the CLI should warn about. However, it doesn't seem super important.